pkg/service/backup/restore_worker_data.go

```diff
@@ -472,6 +477,18 @@ func (w *restoreWorker) newRunProgress(ctx context.Context, run *RestoreRun, tar
 	return pr, nil
 }
 
+func (w *restoreWorker) updateBatchSizeMetric(ctx context.Context, clusterID uuid.UUID, batch []string, host, srcDir string) {
+	var batchSize int64
+	for _, file := range batch {
+		fi, err := w.Client.RcloneFileInfo(ctx, host, srcDir+"/"+file)
+		if err != nil {
+			w.Logger.Error(ctx, "Failed to check file size", "err", err)
+		}
+		batchSize += fi.Size
+	}
```

Comment:
Maybe it's better to fetch this information for all files using `RcloneListDirIter` at the beginning of restoring given table?

Reply:
Moreover, fetching it here will be in conflict with restoring versioned SSTables, because `batch` contains only the regular file names without their specific versions.

Reply:
Let's do it, after you merge SSTable versioning.

Reply:
SSTable versioning is now merged!